Objects/genobject.c

```diff
@@ -409,11 +409,10 @@ gen_close(PyGenObject *gen, PyObject *args)
         return NULL;
     }
     if (PyErr_ExceptionMatches(PyExc_StopIteration)) {
```

Comment:
This check is happening twice now - once here and then again in _PyGen_FetchStopIterationValue.

Maybe this could be instead something like:

```
assert(PyErr_Occurred());
PyObject *value;
if (_PyGen_FetchStopIterationValue(&value) == 0) {
    return value;
}
else if (PyErr_ExceptionMatches(PyExc_GeneratorExit)) {
...
}
```

Reply:
Isn't that basically zero-sum since we don't have to check for success of `_PyGen_FetchStopIterationValue()`?